services/server/src/server/apiv1/verification/private/stateless/customReplaceMethods.ts

```diff
@@ -0,0 +1,164 @@
+import { VerificationExport } from "@ethereum-sourcify/lib-sourcify";
+import { PoolClient } from "pg";
+import {
+  bytesFromString,
+  getDatabaseColumnsFromVerification,
+} from "../../../../services/utils/database-util";
+
+export type CustomReplaceMethod = (
+  poolClient: PoolClient,
+  verification: VerificationExport,
+) => Promise<void>;
+
+export const replaceCreationInformation: CustomReplaceMethod = async (
+  poolClient: PoolClient,
+  verification: VerificationExport,
+) => {
+  if (
+    verification.status.creationMatch !== "perfect" &&
+    verification.status.creationMatch !== "partial"
+  ) {
+    throw new Error(
+      "Creation match is null, cannot replace creation information",
+    );
+  }
+
+  // Get database columns from verification
+  const databaseColumns =
+    await getDatabaseColumnsFromVerification(verification);
+
+  // Get existing verified contract to find deployment_id
+  const existingVerifiedContractQuery = `
+    SELECT vc.id, vc.deployment_id, cd.chain_id, cd.address, cd.contract_id
+    FROM verified_contracts vc
+    JOIN contract_deployments cd ON cd.id = vc.deployment_id
+    LEFT JOIN sourcify_matches sm ON sm.verified_contract_id = vc.id
```

Comment:
Since you use a left join here, am I right that you could get multiple verified_contracts for the same deployment here? Shouldn't we just care about the one referenced by a sourcify_match?

Reply:
Good catch, I actually wanted to use an INNER JOIN